### PR TITLE
Implemented the Storage for the key-value type

### PR DESCRIPTION
**Pull request by aksoli666**
Override the methods put, get, size
Created the class Pair<K, V>

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/Pair.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.impl;
+
+public class Pair<K, V> {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/Pair.java:3]: Make it inner class inside StorageImpl

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_NUM_VALUE = 10;
+    private final Object[] storageArray = new Object[MAX_NUM_VALUE];
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]: image

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,48 @@
...
+    private int size = 0;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,48 @@
...
     public void put(K key, V value) {
+        if (size > MAX_NUM_VALUE) {
+            throw new RuntimeException("Storage is full!");
+        }
+
+        for (int i = 0; i < MAX_NUM_VALUE; i++) {
+            @SuppressWarnings("unchecked")
+            Pair<K, V> pair = (Pair<K, V>) storageArray[i];
+            if (pair != null) {
+                if ((key == null && pair.getKey() == null)
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:20]:
```suggestion
                if (key == pair.getKey() 
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,48 @@
...
+        storageArray[size] = new Pair<>(key, value);
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            @SuppressWarnings("unchecked")
+            Pair<K, V> pair = (Pair<K, V>) storageArray[i];
+            if (pair != null) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:36]: do we need this IF?

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -1,19 +1,54 @@
...
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_NUM_VALUE = 10;
+    private Pair<K, V>[] storageArray;
+    private int size;
+
+    public StorageImpl() {
+        this.storageArray = new Pair[MAX_NUM_VALUE];
+        this.size = 0;
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:13]: int variable is zero by default
```suggestion
```

File: src/main/java/core/basesyntax/Pair.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public V getValue() {
+        return value;
+    }
+
+    public void setValue(V value) {
+        this.value = value;
+    }
+}
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Pair.java:23]: not fixed previous comment

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -1,19 +1,54 @@
 package core.basesyntax.impl;
 
+import core.basesyntax.Pair;
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_NUM_VALUE = 10;
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]: variable name is unclear to understand - what max number value? we have int numbers until 2147483647 
```suggestion
    private static final int MAX_STORAGE_SIZE = 10;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 9]
```diff
@@ -1,19 +1,54 @@
...
+        if (size > MAX_NUM_VALUE) {
+            throw new RuntimeException("Storage is full!");
+        }
+
+        for (int i = 0; i < MAX_NUM_VALUE; i++) {
+            @SuppressWarnings("unchecked")
+            Pair<K, V> pair = storageArray[i];
+            if (pair != null) {
+                if ((key == pair.getKey())
+                        || (key != null && key.equals(pair.getKey()))) {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:27]: not fixed previous comment

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 10]
```diff
@@ -3,17 +3,73 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private Pair<K, V>[] storageArray;
+    private int size;
+
+    public StorageImpl() {
+        this.storageArray = new Pair[MAX_STORAGE_SIZE];
+        this.size = size;
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:12]: you are initializing empty variable with itself? :)
```suggestion
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

**Review by kerrrusha — APPROVED** [review]

**Pull request closed without merging**